lib/Runtime/Base/LeaveScriptObject.cpp

```diff
@@ -63,7 +63,7 @@ namespace Js
             // not run and we might be in an inconsistent state
 
             // Put any code that may raise an exception in OnScriptStart
-            scriptContext->GetThreadContext()->EnterScriptStart(entryExitRecord, doCleanup);
+            scriptContext->GetThreadContext()->EnterScriptStart(entryExitRecord, doCleanup, &this->isScriptActive);
```

Comment:
This is to allow e.g. ImportCall() -> FetchImportedModule() -> InitializeModuleRecord().
When FetchImportedModule() from host calls into chakra, it goes through EnterScript-LeaveScript.
Non-nested EnterScript-LeaveScript resets threadContext->isScriptActive to false.


---
In reply to: [114931889](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r114931889) [](ancestors = 114931889)